libs/designsystem/src/lib/components/chart/chart.component.ts

```diff
@@ -44,152 +57,185 @@ export class ChartComponent implements OnInit, OnChanges {
+        this.mergedOptions = this.donutOptions;
+        this.mergedOptions.chart.type = this.type;
+        this.mergedOptions.plotOptions.pie.innerSize = '0%';
         break;
       }
       case ChartType.AREASPLINE: {
-        this.options = this.areasplineOptions;
-        this.options.chart.type = this.type;
+        this.mergedOptions = this.areasplineOptions;
+        this.mergedOptions.chart.type = this.type;
         break;
       }
       case ChartType.TIMESERIES: {
-        this.options = this.timeSeriesOptions;
-        this.options.chart.type = this.type;
+        this.mergedOptions = this.timeSeriesOptions;
+        this.mergedOptions.chart.type = this.type;
         break;
       }
       case ChartType.ACTIVITYGAUGE: {
-        this.options = this.activitygaugeOptions;
-        this.options.chart.type = this.type;
+        this.mergedOptions = this.activitygaugeOptions;
+        this.mergedOptions.chart.type = this.type;
         break;
       }
       case ChartType.COLUMN: {
-        this.options = getColumnOptions(this.data, this.categories);
-        this.options.chart.type = this.type;
+        this.mergedOptions = columnOptions;
+        this.mergedOptions.chart.type = this.type;
         break;
       }
       case ChartType.BAR: {
-        this.options = getBarOptions(this.data, this.categories);
-        this.options.chart.type = this.type;
+        this.mergedOptions = barOptions;
+        this.mergedOptions.chart.type = this.type;
         break;
       }
     }
   }
 
   updateProperties() {
-    if (this.options.chart) {
-      this.options.chart.height = this.height;
-      this.options.accessibility.description = this.description;
-      switch (this.options.chart.type) {
-        case ChartType.PIE:
-          (this.options.plotOptions.pie
-            .dataLabels as PlotSeriesDataLabelsOptions).enabled = this.showDataLabels;
-        /* falls through */
-        case ChartType.DONUT: {
-          this.options.series = [
-            {
-              type: 'pie',
-              data: this.data,
-            },
-          ];
-          break;
-        }
-        case ChartType.AREASPLINE: {
-          this.options.series = [
-            {
-              type: 'areaspline',
-              data: this.data,
-            },
-          ];
-          break;
-        }
-        case ChartType.TIMESERIES: {
-          this.options.series = [
-            {
-              type: 'area',
-              data: this.data,
-            },
-          ];
-          this.options.xAxis = {
-            ...this.options.xAxis,
-            breaks: this.breaks,
-          };
-          break;
-        }
-        case ChartType.ACTIVITYGAUGE: {
-          const data = this.data[0];
+    this.mergedOptions.chart.height = this.height;
+    this.mergedOptions.accessibility.description = this.description;
+    switch (this.mergedOptions.chart.type) {
+      case ChartType.PIE:
+        this.setPieInput();
+      /* falls through */
```

Comment:
Don't we need to do:
 `this.setSeries('pie');`
?

Reply:
Yes, it "falls through" and uses the donut case as well which does setSeries('pie')